src/components/input/inputDate/inputDate.tsx

```diff
@@ -0,0 +1,41 @@
+import classNames from 'classnames';
+import { forwardRef, useRef } from 'react';
+import { mergeRefs } from 'react-merge-refs';
+import { Button } from '../../button';
+import { IconType } from '../../icon';
+import { InputContainer, type IInputComponentProps } from '../inputContainer';
+import { useInputProps } from '../useInputProps';
+
+export interface IInputDateProps extends IInputComponentProps {}
+
+export const InputDate: React.FC<IInputDateProps> = forwardRef((props, ref) => {
+    const { containerProps, inputProps } = useInputProps(props);
+
+    const { className: inputClassName, ...otherInputProps } = inputProps;
+
+    const inputRef = useRef<HTMLInputElement>(null);
+
+    const handleCalendarClick = () => {
+        inputRef.current?.showPicker();
+    };
+
+    return (
+        <InputContainer {...containerProps}>
+            <input
+                type="date"
+                className={classNames('calendar-icon:hidden calendar-icon:appearance-none', inputClassName)}
+                ref={mergeRefs([inputRef, ref])}
+                {...otherInputProps}
+            />
+            <Button
```

Comment:
Don't forget to pass in the `disabled` prop down to the button.

Reply:
Good catch, adding a test as well 👍🏻